ambry-store/src/test/java/com.github.ambry.store/StorageManagerTest.java

```diff
@@ -164,11 +164,11 @@ public void storeStartFailureOnOneDiskTest() throws Exception {
 
   /**
    * Test that stores for all partitions on a node have been started and partitions not present on this node are
-   * inaccessible.
+   * inaccessible. Also tests all stores are shutdown on shutting down the storage manager
    * @throws Exception
    */
   @Test
-  public void successfulStartTest() throws Exception {
+  public void successfulStartupShutdownTest() throws Exception {
```

Comment:
I think you should also test the metric that tracks shutdown failure count to ensure it is 0 in that function. Stores that haven't been started at all should not be throwing exceptions and incrementing the counter for e.g.